storage/merkle/proof_test.go

```diff
@@ -0,0 +1,166 @@
+package merkle
+
+import (
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// TestProofWithASingleKey tests proof generation and verification
+// when trie includes only a single value
+func TestProofWithASingleKey(t *testing.T) {
```

Comment:
We should be testing boundary conditions here, i.e. keys with the **smallest possible size** and keys with the **largest possible size**. In addition, I would also test with some key in the middle (like case you already have).

Reply:
good suggestion, updated the test to cover all cases